src/main/java/dev/waterdog/utils/Configuration.java

```diff
@@ -59,91 +83,147 @@ public void loadFrom(Map<String, Object> values) {
     }
 
     public Set<String> getKeys() {
-        return new HashSet<>(this.values.keySet());
+        return this.values.keySet();
+    }
+
+    public void remove(String key) {
+        LastMap lastMap = getLastMap(key);
+
+        if (lastMap == null) return;
+
+        lastMap.map.remove(lastMap.key);
     }
 
     public void set(String key, Object value) {
-        this.values.put(key, value);
+        LastMap lastMap = getLastMap(key);
+
+        if (lastMap == null) return;
+
+        lastMap.map.put(lastMap.key, value);
+    }
+
+    @SuppressWarnings("unchecked")
+    private LastMap getLastMap(String key) {
+        if (key == null || key.isEmpty()) return null;
+
+        Map<String, Object> values = this.values;
+        String[] keys = key.split("\\.");
+        String currentKey = null;
+
+        for (int i = 0; i < keys.length; i++) {
+            currentKey = keys[i];
+
+            if (i + 1 < keys.length && values.get(currentKey) == null) {
+                values.put(currentKey, new LinkedHashMap<>());
```

Comment:
Whats the reason behind using `LinkedHashMap` here?

Reply:
Organization.